builtin/providers/azurerm/resource_arm_express_route_circuit_test.go

```diff
@@ -0,0 +1,114 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"testing"
+
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMExpressRouteCircuit_basic(t *testing.T) {
+	var erc network.ExpressRouteCircuit
+	ri := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMExpressRouteCircuitDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMExpressRouteCircuit_basic(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMExpressRouteCircuitExists("azurerm_express_route_circuit.test", &erc),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMExpressRouteCircuitExists(name string, erc *network.ExpressRouteCircuit) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		expressRouteCircuitName := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Express Route Circuit: %s", expressRouteCircuitName)
+		}
+
+		conn := testAccProvider.Meta().(*ArmClient).expressRouteCircuitClient
+
+		resp, err := conn.Get(resourceGroup, expressRouteCircuitName)
+		if err != nil {
+			if resp.StatusCode == http.StatusNotFound {
+				return fmt.Errorf("Bad: Express Route Circuit %q (resource group: %q) does not exist", expressRouteCircuitName, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get on expressRouteCircuitClient: %s", err)
+		}
+
+		*erc = resp
+
+		return nil
+	}
+}
+
+func testCheckAzureRMExpressRouteCircuitDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).expressRouteCircuitClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_express_route_circuit" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(resourceGroup, name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Express Route Circuit still exists:\n%#v", resp.ExpressRouteCircuitPropertiesFormat)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMExpressRouteCircuit_basic(rInt int) string {
+	return strings.Replace(`
```

Comment:
@whiskeyjay so one option to avoid that issue when using a `fmt.Sprintf` is to use the substitute value of `%[1]d`to reference the first item rather than `%d` for this purpose (as you've seen we've started wrapping the test strings in a method, rather than formatting the string inside of each test to avoid the argument mismatch issue). 

But yes, it'd be nice to make this consistent with the other acceptance tests if possible :)